RELEASE_NOTES.rst

```diff
@@ -230,6 +230,9 @@ Target version for core dependency ``pendulum`` package set to 3 (#36281).
 Support for pendulum 2.1.2 will be saved for a while, presumably until the next feature version of Airflow.
 It is advised to upgrade user code to use pendulum 3 as soon as possible.
 
+Pendulum 3 introduced some subtle incompatibilities that you might rely on in your code - for example
+default rendering of ISO8601 dates is missing ``T`` in the rendered date.
```

Comment:
Maybe worth to mention that this is a default `datetime.datetime` str behaviour (pendulum.DateTime subclass it), and if someone relay on that, better use [`.isoformat`](https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat) method

Reply:
yep. good point

Reply:
updated.